### PR TITLE
Set progressing status just before trying to create SC

### PR DESCRIPTION
**Pull request by wongma7**
fix bz 1688582

### CODE REVIEW

**Comment by wongma7** [comment]
@openshift/storage pls review.
operator is supposed to set Progressing before available/failing. otherwise user does not see 'lastTransitionTime' as there is no transition available->available, even though the version may actually have changed

File: pkg/controller/clusterstorage/clusterstorage_controller.go [hunk 1]
```diff
@@ -186,7 +192,58 @@ func (r *ReconcileClusterStorage) Reconcile(request reconcile.Request) (reconcil
...
+	notProgressing = configv1.ClusterOperatorStatusCondition{
+		Type:   configv1.OperatorProgressing,
+		Status: configv1.ConditionFalse,
+	}
+)
+
+// setStatusProgressing sets Available=false;Failing=false;Progressing=true
+// we set "progressing" if the cluster operator's version is not the latest
+// and we are about to try to roll it out
+func (r *ReconcileClusterStorage) setStatusProgressing(clusterOperator *configv1.ClusterOperator) error {
```
Comment (gnufied) [inline pkg/controller/clusterstorage/clusterstorage_controller.go]: Can we write some unit tests for this stuff?

Comment (wongma7) [reply pkg/controller/clusterstorage/clusterstorage_controller.go]: done. will write one for syncStatus in follow up

**Comment by gnufied** [comment]
Failures here look unrelated. (storage flakes... :( )

/lgtm

**Pull request merged**